### PR TITLE
Make RocksJava compliant with Java 8 language level (EB 1046)

### PR DESCRIPTION
**Pull request by alanpaxton**
A systematic attempt to apply a set of IntelliJ Java inspections to the Java interface to RocksDB and its tests.

The public interface was left unmodified, except for removal of unchecked exceptions from the explicit signatures.
Parameters were made final where possible.

Individual suppressions were introduced where the effort to change code contrary to the defined inspections was too great. We are left in a position where new and modified code can be maintained with passing inspections.

Some tests which previously used other assertion models were converted to assertj, e.g. `(assertThat(actual).isEqualTo(expected)`

In a very few cases tests were found to be inoperative or broken, and were repaired. No problems with actual RocksDB behaviour were observed.

A number of other standard transformations were applied per the hints or quick-fixes of the defined inspections. Again, these should not have affected the interface and should always have been behaviour preserving on the code.

### CODE REVIEW

**Review by jay-zhuang — APPROVED** [review]
LGTM

**Comment by alanpaxton** [comment]
> There're several Java tests failed, please check

will do

**Comment by alanpaxton** [comment]
I organised imports in IntelliJ to force another build. This has now succeeded without the errors building `RocksJunitRunner` 
The last set of failures was either an intermittent build failure (S3 problem fetching JARs) or the previous re-ordering of imports from `make format` managed to mask or fail to import some Java names.

**Comment by adamretter** [comment]
@jay-zhuang I think this should be good to go now :-) Thanks @alanpaxton

File: java/src/main/java/org/rocksdb/FlushJobInfo.java [hunk 1]
```diff
@@ -151,19 +160,24 @@ public FlushReason getFlushReason() {
...
+      if (threadId == that.threadId)
+        if (jobId == that.jobId)
+          if (triggeredWritesSlowdown == that.triggeredWritesSlowdown)
+            if (triggeredWritesStop == that.triggeredWritesStop)
+              if (smallestSeqno == that.smallestSeqno)
+                if (largestSeqno == that.largestSeqno)
+                  if (Objects.equals(columnFamilyName, that.columnFamilyName))
+                    if (Objects.equals(filePath, that.filePath))
+                      if (Objects.equals(tableProperties, that.tableProperties))
+                        return flushReason == that.flushReason;
```
Comment (jay-zhuang) [inline java/src/main/java/org/rocksdb/FlushJobInfo.java:179]: I'm not sure if this is better than the older style.

File: java/src/main/java/org/rocksdb/FlushJobInfo.java [hunk 2]
```diff
@@ -175,12 +189,9 @@ public int hashCode() {
...
   public String toString() {
-    return "FlushJobInfo{"
-        + "columnFamilyId=" + columnFamilyId + ", columnFamilyName='" + columnFamilyName + '\''
-        + ", filePath='" + filePath + '\'' + ", threadId=" + threadId + ", jobId=" + jobId
-        + ", triggeredWritesSlowdown=" + triggeredWritesSlowdown
-        + ", triggeredWritesStop=" + triggeredWritesStop + ", smallestSeqno=" + smallestSeqno
-        + ", largestSeqno=" + largestSeqno + ", tableProperties=" + tableProperties
-        + ", flushReason=" + flushReason + '}';
+    return MessageFormat.format(
+        "FlushJobInfo'{'columnFamilyId={0}, columnFamilyName=''{1}'', filePath=''{2}'', threadId={3}, jobId={4}, triggeredWritesSlowdown={5}, triggeredWritesStop={6}, smallestSeqno={7}, largestSeqno={8}, tableProperties={9}, flushReason={10}'}'",
```
Comment (jay-zhuang) [inline java/src/main/java/org/rocksdb/FlushJobInfo.java:193]: nit: is this line too long.

File: java/src/main/java/org/rocksdb/util/package-info.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+/**
```
Comment (jay-zhuang) [inline java/src/main/java/org/rocksdb/util/package-info.java:1]: please add LICENSE header.

File: java/src/test/java/org/rocksdb/test/RocksJunitRunner.java [hunk 4]
```diff
@@ -32,20 +31,18 @@
...
 
-    private String currentClassName = null;
-    private String currentMethodName = null;
-    private Status currentStatus = null;
-    private long currentTestsStartTime;
-    private int currentTestsCount = 0;
-    private int currentTestsIgnoredCount = 0;
-    private int currentTestsFailureCount = 0;
-    private int currentTestsErrorCount = 0;
+    @SuppressWarnings("RedundantFieldInitialization") private String currentClassName = null;
```
Comment (jay-zhuang) [inline java/src/test/java/org/rocksdb/test/RocksJunitRunner.java:38]: Is this really redundant field initialization? if so maybe we should remove the initialization.

File: java/src/test/java/org/rocksdb/test/package-info.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+/**
+ * This module contains tests for the Java interface to the RocksDB key,value store
+ * <p>
+ * </p>
+ *
+ * @author alanpaxton
+ * @version 1.0
+ */
+package org.rocksdb.test;
```
Comment (jay-zhuang) [inline java/src/test/java/org/rocksdb/test/package-info.java:9]: new line at the end of the file please.

File: java/src/test/java/org/rocksdb/test/package-info.java [hunk 6]
```diff
@@ -0,0 +1,9 @@
+/**
```
Comment (jay-zhuang) [inline java/src/test/java/org/rocksdb/test/package-info.java:1]: license header please.

File: java/src/main/java/org/rocksdb/package-info.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+/**
```
Comment (jay-zhuang) [inline java/src/main/java/org/rocksdb/package-info.java:1]: license header please.

**Review by jay-zhuang — APPROVED** [review]
LGTM

**Review by japui-coder — COMMENTED** [review]
java/src/main/java/org/rocksdb/FlushJobIjava/src/main/java/org/rocksdb/FlushJobInfo.javanfo.java . Patch java/src/main/java/org/rocksdb/FlushOptions.java .  Patch  .

**Review by japui-coder — COMMENTED** [review]
java/inspectionProfiles/Project_Default.xml . Approved .

**Review by japui-coder — COMMENTED** [review]
java/src/main/java/org/rocksdb/util/Environment.java . Patch.

**Review by japui-coder — COMMENTED** [review]
java/src/test/java/org/rocksdb/ColumnFamilyTest.java . Patch.

**Review by japui-coder — COMMENTED** [review]
java/src/main/java/org/rocksdb/RocksDB.java . Patch.

**Review by japui-coder — COMMENTED** [review]
java/src/main/java/org/rocksdb/RocksDB.java . Patch.

**Comment by alanpaxton** [comment]
This PR covers too much stuff and has generated a bunch of conflicts while waiting for review. I've replaced it with the less ambitious https://github.com/facebook/rocksdb/pull/10951 and will close this.

**Pull request closed without merging**